include/RAJA/internal/LegacyCompatibility.hpp

```diff
@@ -186,12 +164,12 @@ RAJA_HOST_DEVICE RAJA_INLINE constexpr auto foldl(Op&& operation,
                                                   Rest&&... rest) ->
     typename foldl_impl<Op, Arg1, Arg2, Arg3, Rest...>::Ret
 {
-  return foldl(forward<Op&&>(operation),
-               forward<Op&&>(
-                   operation)(forward<Op&&>(operation)(forward<Arg1&&>(arg1),
-                                                       forward<Arg2&&>(arg2)),
-                              forward<Arg3&&>(arg3)),
-               forward<Rest&&>(rest)...);
+  return foldl(camp::forward<Op&&>(operation),
```

Comment:
I believe the additional `&&` in the template is unnecessary. The arguments are already specified as universal references and reference collapsing rules would keep all of the types the same.

Reply:
Edit: with `std::forward` that is the case. `camp::forward` this may be required.

Reply:
You were right in the first place, this should be just Op.